### ISSUE TITLE
Regression: Parsing a field of type float will result in a date parse failure

### ISSUE THREAD

**Issue by ph** [logstash-plugins/logstash-filter-date#84]
This was reported by @msimos 

When you use the **JSON filter** for parsing a json string and use a parsed field of the float type it will make the date filter fails.  Converting the field to a string with the mutate will make the parsing succeed.

**Sample json**
```
{"ts":1482283085.15707,"uid":"C5yKv433r1FNFeQ8C1" }
{"ts":1482284384.052604, "uid":"CZbhyyq9n7dE5gg1h" }
```
**Config**
```
input {
  file {
      path => ["/Users/ph/Downloads/logstash-5.1.1/conn.log"]
      sincedb_path => "/dev/null"
      start_position => "beginning"
  }
}
filter {
    json {
        source => "message"

    }
    date {
        match => ["ts", "UNIX" ]
    }
}
output {
    stdout { codec => rubydebug }
}
```

**Results**
```
{
          "path" => "/Users/ph/Downloads/logstash-5.1.1/conn.log",
           "uid" => "CZbhyyq9n7dE5gg1h",
    "@timestamp" => 2016-12-21T21:28:33.928Z,
      "@version" => "1",
          "host" => "sashimi",
       "message" => "{\"ts\":1482284384.052604, \"uid\":\"CZbhyyq9n7dE5gg1h\" }",
          "tags" => [
        [0] "_dateparsefailure"
    ],
            "ts" => 1482284384.052604
}
```

Another interesting issue if we use the json codec instead of using the json filter the date filter can parse the field. I am wondering if this is not related to the BiValues problem instead of the date filter? I need to look more at the implementation.

**It work with 2.4**

cc @guyboertje

**Comment by guyboertje** [comment]
@ph I suspect you are correct.
In the JSON codec the code path uses the pure Java `fromJson` method which will create a DoubleBiValue and the Date filter has a Java converter method for Double.
In the JSON filter, the code path uses JrJackson and a BigDecimalBiValue will be created and the Date filter does not know how to handle that.

This is the risk we took by having two different JSON processor implementations.

We should extend the Date filter to handle BigDecimal.

**Comment by ph** [comment]
I believe BigDecimal is the way to go to.
My guts where pointing at BiValues. :)

Also added the note about 2.4.

**Comment by jacobweber** [comment]
I have this exact problem, but converting the floats to strings didn't work, since it resulted in strings like "0.14761403301869E10" (when the float was 1476140332.1238). The date parser doesn't fail, but it returns a timestamp like 1970-01-01T00:00:00.147Z.

Anyone know a workaround?

**Comment by guyboertje** [comment]
Fixed by #89
